src/components/timecard/edit-shift/EditShift.jsx

```diff
@@ -0,0 +1,509 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import { UrlSearchParamBuilder } from '../../../utils/api-utils/UrlSearchParamBuilder';
+import {
+  formatDate,
+  formatDateTimeISO,
+  formatTime,
+  removeTimecardContextEntry,
+  isFinishTimeOnNextDay,
+} from '../../../utils/time-entry-utils/timeEntryUtils';
+import { ContextTimeEntry } from '../../../utils/time-entry-utils/ContextTimeEntry';
+import { focusErrors } from '../../../utils/common-utils/common-utils';
+import { validateServiceErrors } from '../../../utils/api-utils/ApiUtils';
+import { useEffect, useState } from 'react';
+import { clashingProperties, inputNames } from '../../../utils/constants';
+import { combineExistingAndTimeClashErrors } from '../../../utils/time-entry-utils/combineTimeErrors';
+import StartFinishDateInput from '../start-finish-date-input/StartFinishDateInput';
+import Checkbox from '../../common/form/checkbox/Checkbox';
+import { sortErrors } from '../../../utils/sort-errors/sortErrors';
+
+const EditShift = ({
+  summaryErrors,
+  setSummaryErrors,
+  timecardDate,
+  setShowEditShiftHours,
+  timeEntry,
+  timeEntriesIndex,
+  timeEntries,
+  setTimeEntries,
+  setSummaryMessages,
+  timePeriodTypesMap,
+}) => {
+  const { register, handleSubmit, formState, getValues } = useForm({
+    reValidateMode: 'onSubmit',
+    shouldFocusError: false,
+  });
+
+  const inputName = 'shift';
+  const { setServiceError, userId } = useApplicationContext();
+  const [isChecked, setIsChecked] = useState(false);
+  const [clashingProperty, setClashingProperty] = useState(null);
+  const [clashingTimes, setClashingTimes] = useState(null);
+
+  const startEntryExists = !!timeEntry?.startTime && timeEntry.startTime !== '';
+  const finishEntryExists =
+    !!timeEntry?.finishTime && timeEntry.finishTime !== '';
+  const [localStartDate, setLocalStartDate] = useState(
+    startEntryExists ? timeEntry.startTime : timecardDate
+  );
+  const [localEndDate, setLocalEndDate] = useState(
+    finishEntryExists ? timeEntry.finishTime : timecardDate
+  );
+
+  const [finishTimeText, setFinishTimeText] = useState('');
+
+  useEffect(() => {
+    focusErrors(document.querySelector('[id^="summary-error"] a'));
+  }, [summaryErrors]);
+
+  const handleCheckboxChange = () => {
+    setIsChecked((isChecked) => !isChecked);
+  };
+
+  const handleError = (errorFields) => {
+    setSummaryErrors(errorFields);
+  };
+
+  const updateFinishTimeText = (startDate, endDate) => {
+    const actualStartDate = startDate ? startDate : localStartDate;
+    const actualEndDate = endDate ? endDate : localEndDate;
+
+    if (!isChecked) {
+      let startTime = getValues(inputNames.shiftStartTime);
+      let finishTime = getValues(inputNames.shiftFinishTime);
+      if (!startTime) {
+        startTime = formatTime(timeEntry.startTime);
+      }
+      if (!finishTime) {
+        finishTime = formatTime(timeEntry.finishTime);
+      }
```

Comment:
This seems confusing - might need to jump on a call to go through this bit

Reply:
Discussed with @IanFisherHO and @LaurieBennett - this will be fixed in the dynamic shift hours ticket